src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -813,26 +845,20 @@ bool QgsMapBoxGlStyleConverter::parseCircleLayer( const QVariantMap &jsonLayer,
   symbol->setOutputUnit( context.targetUnit() );
   symbol->setDataDefinedProperties( ddProperties );
 
-  if ( circleOpacity != -1 )
+  if ( !circleTranslate.isNull() )
   {
-    symbol->setOpacity( circleOpacity );
-
-    if ( circleStrokeOpacity != -1 )
-    {
-      context.pushWarning( QObject::tr( "%1: Could not set opacity of layer, opacity already defined in fill color" ).arg( context.layerId() ) );
-    }
-  }
-  else if ( circleStrokeOpacity != -1 )
-  {
-    symbol->setOpacity( circleStrokeOpacity );
+    markerSymbolLayer->setOffset( circleTranslate );
+    markerSymbolLayer->setOffsetUnit( context.targetUnit() );
   }
+
   if ( circleFillColor.isValid() )
   {
     markerSymbolLayer->setFillColor( circleFillColor );
   }
   if ( circleRadius != -1 )
   {
-    markerSymbolLayer->setSize( circleRadius );
+    markerSymbolLayer->setSize( 2 * circleRadius ); // diameter
```

Comment:
This won't fix data defined interpolated sizes -- to do that you need to move the scaling factor of 2 up to the earlier block like in my initial review.

Reply:
ah... now I understand :)